### PR TITLE
Strict null checking

### PR DESCRIPTION
**Pull request by zwzn**

### CODE REVIEW

File: web/src/js/views/series-view.tsx [hunk 1]
```diff
@@ -30,20 +30,24 @@ export default class SeriesView extends Component<Props, State> {
         this.seriesChange()
     }
     public async componentDidUpdate(prevProps: Props) {
+<<<<<<< HEAD
```
Comment (buckley-w-david) [inline web/src/js/views/series-view.tsx]: I spy with my little eye something that looks like a merge conflict

Comment (zwzn) [reply web/src/js/views/series-view.tsx]: I forgot to save... woops

Comment (buckley-w-david) [reply web/src/js/views/series-view.tsx]: RIP

File: web/src/js/views/settings.tsx [hunk 2]
```diff
@@ -46,11 +46,11 @@ export default class Settings extends Component<{}, State> {
                 name: me.name,
             })
         })
-        Options.getOrigin().then(address => {
-            this.setState({
-                address: address,
-            })
-        })
+        // Options.getOrigin().then(address => {
```
Comment (buckley-w-david) [inline web/src/js/views/settings.tsx]: I'm not digging the commented out code, are we expecting to un-comment it in the near future? If not it should likely just be removed if we don't want it.

Comment (zwzn) [reply web/src/js/views/settings.tsx]: That was my pre git coding coming through. I'll remove it

File: web/src/js/options.ts [hunk 3]
```diff
@@ -2,26 +2,26 @@ import { gql } from 'js/graphql'
...
-    }
-    public async setOrigin(o?: string): Promise<string> {
-        const oldOrigin = await this.getOrigin()
-        try {
-            localStorage.setItem('origin', o)
-            await gql('me{id}')
-            return o
-        } catch (e) {
-            const origin = prompt('what origin are you useing? e.g. "http://192.168.1.10:8080"')
+    // public async getOrigin(): Promise<string> {
```
Comment (buckley-w-david) [inline web/src/js/options.ts]: :fire:

File: web/src/js/options.ts [hunk 4]
```diff
@@ -2,26 +2,26 @@ import { gql } from 'js/graphql'
...
+    //     return localStorage.getItem('origin') \
```
Comment (buckley-w-david) [inline web/src/js/options.ts]: :fire:

File: web/src/js/options.ts [hunk 5]
```diff
@@ -2,26 +2,26 @@ import { gql } from 'js/graphql'
...
 
-            if (origin) {
-                return await this.setOrigin(origin)
-            } else {
-                localStorage.setItem('origin', oldOrigin)
-                return await this.getOrigin()
-            }
-        }
-    }
+    //         if (origin) {
```
Comment (buckley-w-david) [inline web/src/js/options.ts]: :fire:

File: web/src/js/url.ts [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
 export default async function url(path: string): Promise<string> {
-    if (location.protocol === 'file:') {
-        let origin = await Options.getOrigin()
-        if (!origin) {
-            origin = prompt('what origin are you useing? e.g. "http://192.168.1.10:8080"')
-            await Options.setOrigin(origin)
-        }
-        return origin + path
-    }
+    // if (location.protocol === 'file:') {
```
Comment (buckley-w-david) [inline web/src/js/url.ts]: :fire:

File: web/src/js/views/settings.tsx [hunk 7]
```diff
@@ -224,10 +218,10 @@ export default class Settings extends Component<{}, State> {
 
     @autobind
     private async btnAddress() {
-        const address = await Options.setOrigin()
-        this.setState({
-            address: address,
-        })
+        // const address = await Options.setOrigin()
```
Comment (buckley-w-david) [inline web/src/js/views/settings.tsx]: :fire:

**Comment by buckley-w-david** [comment]
image

So when attempting to test the branch I get that ^

**Comment by zwzn** [comment]
My bad, I needed to have an empty JSON object if the local storage wasn't set, not an empty string

**Comment by buckley-w-david** [comment]
peek 2018-12-10 15-48

Book transitions are broken, now it just loops

**Comment by zwzn** [comment]
I think it should work now.

File: web/src/js/views/read.tsx [hunk 8]
```diff
@@ -228,27 +230,26 @@ export default class Read extends Component<Props, State> {
...
-        book.then(b => {
-            if (b != null) {
-                route('/book/' + b.id)
-                this.initBook(b)
-            } else {
-                route('/series/' + bk.series)
-            }
-        })
+        const book = await query.first()
+console.log(book);
```
Comment (buckley-w-david) [inline web/src/js/views/read.tsx]: :fire:

Comment (buckley-w-david) [reply web/src/js/views/read.tsx]: Whoops, was looking at a previous commit, not the most up to date one

**Comment by zwzn** [comment]
Woop

**Pull request merged**